dbms/src/Core/Block.cpp

```diff
@@ -514,6 +514,36 @@ static ReturnType checkBlockStructure(const Block & lhs, const Block & rhs, cons
     return ReturnType(true);
 }
 
+Block mergeBlocks(Blocks && blocks)
+{
+    assert(!blocks.empty());
+    auto & sample_block = blocks[0];
+    size_t result_rows = 0;
+    for (const auto & block : blocks)
+    {
+        result_rows += block.rows();
+    }
+
+    MutableColumns dst_columns(sample_block.columns());
```

Comment:
Can initialize dst_columns with nullptrs here. And can choose a better name instead of 'sample_block' here.